### PR TITLE
[Hack week] Create a new product using images

### LINKED ISSUE #9281: Add option to share images into WooCommerce app as new product
_Please submit all enhancements via the app, using the "Send Feedback" section in settings._ (I could not find this option)

In my business It is very common to receive images for new/existing products for my store.

It could have been amazing to be able to share an image from whatsapp (or gallery or other apps) directly into WooCommerce app and adding a new product from it/adding new image to existing product.

please consider this new feature.

and thanks a lot for this amazing app!

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #9281. This PR adds the ability to share images from different apps and launch the Woo app, which automatically creates a new product using the shared images. The feature was requested by an existing user.

https://github.com/woocommerce/woocommerce-android/assets/1522856/9e1f3446-b7bd-4b0f-bab1-6e2336c4d74b

**To test:**
1. Open an app that can share an image (like a gallery app)
2. Select one ore more images and share them
3. Select the WooCommerce app from the share dialog
4. Notice the WooCommerce app is launched and the selected images are automatically added to a new product and uploaded

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -321,17 +331,18 @@ class MainActivity :
 
         if (savedInstanceState == null) {
             viewModel.handleIncomingAppLink(intent?.data)
-            viewModel.handleShortcutAction(intent?.action?.toLowerCase())
+            viewModel.handleShortcutAction(intent?.action?.toLowerCase(Locale.ROOT))
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:334]: Question for learning: What's the reason for passing Locale.ROOT here?

**Review by hafizrahman — APPROVED** [review]
Love this feature so much! Glad we're adding this to the app. It probably helps a lot with some merchant's workflow especially when they have a lot of product images, or edit a lot in their app.

This worked nicely in my test and the code looks great and clean (just added a non-blocking question).

For future idea we can also support sharing & adding image into an existing product as well.

A suggestion is to also add some rudimentary tracking to see how much this feature gets used, but it's also not a blocker imo.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:334]: That's the default locale-insensitive option when you just want to convert the string to lowercase.

From documentation:

>  The root locale is the one with empty ("") strings for language, country, and variant attributes that define a Locale. This is the foundation locale for all locales, and it is utilized as the language/country-neutral locale for actions that are sensitive to locales.

**Comment by 0nko** [comment]
Thanks, @hafizrahman!

> For future idea we can also support sharing & adding image into an existing product as well.

Yeah, although that would be a bit more work, because you'd need some way to select the product.

> A suggestion is to also add some rudimentary tracking to see how much this feature gets used, but it's also not a blocker imo.

Good call. I've added the tracking.

**Pull request merged**